puppet/zulip/manifests/app_frontend.pp

```diff
@@ -1,6 +1,6 @@
 # Default configuration for a Zulip app frontend
 class zulip::app_frontend {
-  include zulip::common
+  include zulip::base
```

Comment:
Is the idea here that following this series, `zulip::base` is only included from `::profile::` manifests?  Might be worth adding a comment at the top of `zulip::base` explaining that and to use `zulip::common` for variables needed by non-profile rules.  (And maybe a linter check to prevent importing `::profile::` rules in the wrong places?)

Also is this line a violation of the `profile` abstraction layering?
```
puppet/zulip/manifests/app_frontend_base.pp:  $queues_multiprocess = $zulip::profile::base::total_memory_mb > 3500
```

Reply:
> Is the idea here that following this series, zulip::base is only included from ::profile:: manifests?

Yeah; it was talked about in b07d9e05f1f6a0fd49a696566ca87eb67202fe5e a bit, but that change is before the `::profile::` addition, so it was hard to talk about there.  I'll adjust the wording in 93687c0c462a77170bb114c85c1892cbe8262bc8 which introduces `::profile::`, as well as the docs in `zulip::base`.  Linter is a good call.

> Also is this line a violation of the `profile` abstraction layering?
> 
> ```
> puppet/zulip/manifests/app_frontend_base.pp:  $queues_multiprocess = $zulip::profile::base::total_memory_mb > 3500
> ```

Whoops, yeah.  I cleaned most of the vars out of there, but apparently not all.  Will fix and add a lint.

Reply:
Sounds good.  I think it's reasonable in a commit to say "This is preparation for moving all the top-level rules to the zulip::profile:: namespace" or similar, even if you're not doing that until a later commit.